app/models/user.rb

```diff
@@ -404,23 +404,35 @@ def basic_profile_present?
   end
 
   ###Helpers############
-  def self.build(opts = {})
+  def self.build(opts={})
     u = User.new(opts.except(:person, :id))
     u.setup(opts)
     u
   end
 
+  def self.find_or_build(opts={})
+    user = User.find_or_create_by(username: opts[:username], email: opts[:email])
```

Comment:
This is (also) called from the rake file that has no user-context, but in command line a target username must be given. Mail-Address then is read from the import file. 

Maybe this behavior should be well defined and then fixed or documented somewhere.

Reply:
I'm fine with reading the mail-address from the archive, but it shouldn't overwrite the existing mail-address and especially it should still work if the mail-addresses are different. So basically only set the mail-address if it's otherwise missing/empty.

Reply:
Related, the import currently fails if the e-mail in the archive is already taken by another account in the pod. What do we want to do with that?

Reply:
I've already changed this. Only in case a migration is started from a rake file. Its now searched only for the username. In case a username was not found, one is created with attached email.
In case of an online migration, this code is not called, user exists anyway. Username and mail provided in the archive is ignored. 

But what if - a user must be created and the email address is already occupied?

Reply:
>In case a username was not found, one is created with attached email

By attached email you mean the one in the archive right? This is the problem here, it will fail if another user already uses that email, as the  same e-mail can't be used twice.

Reply:
There is not much more you can do about it than just show the error. If a user already has an account with the same mail-address on that pod, then they either need to import it into their existing user, or create a second account with a different mail-address. It's the same as if you just register normally, if the user wants to register with the same mail-address twice it fails.

Reply:
Process fails fast in this case.

Reply:
>There is not much more you can do about it than just show the error

Well, that's the question. Should we block the users who's trying to import the archive saying that this e-mail address is already taken (which means they either have to log in the other account and modify the address, or modify the JSON, which is a pain, we don't want that) or should we just ignore the value of the email in the archive and do the import anyway? I honestly don't know.

Reply:
If the user imports the archive into an existing account, the mail-address is already ignored (at least it should be). It only fails if you import the archive using the rake task into a non-existing account, then creation of the account fails, because there is already another account with the same mail-address. So either use the rake-task to import it into the other account existing account or create a new account first with a different email-address.

Reply:
Exactly.